websites/R/RTLplay/presence.ts

```diff
@@ -0,0 +1,416 @@
+				/* HUB CHANNEL PAGE (Page HUB d'une chaîne)
+
+				(ex: https://www.rtlplay.be/tvi) */
+				default: {
+					presenceData.details = privacy
+						? (await strings).viewAPage
+						: (await strings).viewChannel;
+					presenceData.state = privacy ? "" : `${channel}`;
+
+					presenceData.smallImageKey = Assets.Reading;
+					presenceData.smallImageText = (await strings).browsing;
+
+					if (time) presenceData.startTimestamp = browsingTimestamp;
+
+					break;
+				}
+				/* LIVE CHANNEL PAGE (Page d'une chaîne en direct)
+				aka LIVE VIDEO PLAYER (Player vidéo en direct)
+
+				(ex: https://www.rtlplay.be/tvi/direct) */
+				case pathname.split("/")[2] === "direct":
+					{
+						presenceData.details = privacy
+							? (await strings).watchingLive
+							: `${(await strings).watching} ${channel}`;
+						presenceData.state = privacy
+							? ""
+							: document.querySelector("div.sc-18trp4n-3.bVsrtw > h1")
+									.textContent; // Content of the first line
+
+						presenceData.largeImageKey = getChannel(pathname).logo;
+						presenceData.largeImageText = channel;
+						presenceData.smallImageKey = Assets.Live;
+						presenceData.smallImageText = (await strings).live;
+
+						presenceData.type = getChannel(pathname).type;
+
+						if (time) {
+							if (
+								document
+									.querySelector('[type="duration"]')
+									.textContent.includes("-")
+							) {
+								presenceData.endTimestamp = presence.getTimestamps(
+									0,
+									presence.timestampFromFormat(
+										document
+											.querySelector('[type="duration"]')
+											.textContent.replace("-", "")
+									)
+								)[1];
+							} else {
+								presenceData.endTimestamp = presence.getTimestamps(
+									presence.timestampFromFormat(
+										document.querySelector('[type="currentTime"]').textContent
+									),
+									presence.timestampFromFormat(
+										document.querySelector('[type="duration"]').textContent
+									)
+								)[1];
+							}
+						}
+
+						if (buttons) {
+							presenceData.buttons = [
+								{
+									label: (await strings).buttonWatchStream,
+									url: href, // We are not redirecting directly to the raw stream, it's only the livestream page
+								},
+							];
+						}
+					}
+
+					break;
+			}
+			break;
+		}
+
+		/* MOVIE PAGE & CATEGORY PAGE (Page d'un Film ou d'une Catégorie)
+	There's a little difference between a Movie page and a Category page.
+	The first one has a Play video button that the other one hasn't
+
+	(ex: https://www.rtlplay.be/chicago-fire-p_8947 and https://www.rtlplay.be/rtl_play/series-rtl-play-f_409) */
+		case exist(
+			"header > div > div > h1, div.sc-nqi6nw-0.cNDqJj > h1" // Movie name or Category name: Supports mobile and desktop layout
+		): {
+			presenceData.state = privacy
+				? ""
+				: document.querySelector(
+						"header > div > div > h1, div.sc-nqi6nw-0.cNDqJj > h1" // Movie name or Category name: Supports mobile and desktop layout
```

Comment:
If possible, use selectors which use avoid computer-generated classes or selectors.

Reply:
Thanks for the feedback, I'll look into that !